optaplanner-core/src/main/java/org/optaplanner/core/impl/partitionedsearch/DefaultPartitionedSearchPhase.java

```diff
@@ -184,10 +185,8 @@ private ExecutorService createThreadPoolExecutor(int partCount) {
             phaseList.add(phase);
             partPhaseIndex++;
         }
-        // TODO create PartitionSolverScope alternative to deal with 3 layer terminations
-        SolverScope<Solution_> partSolverScope = solverScope.createChildThreadSolverScope(ChildThreadType.PART_THREAD);
-        partSolverScope.setRunnableThreadSemaphore(runnablePartThreadSemaphore);
-        return new PartitionSolver<>(bestSolutionRecaller, partTermination, phaseList, partSolverScope);
+        return new PartitionSolver<>(bestSolutionRecaller, partTermination, phaseList, solverScope,
+                runnablePartThreadSemaphore);
```

Comment:
`SolverScope` now carries a reference to the `Solver`, therefore it is created within the solvers, not externally.